routing/active_payment.go

```diff
@@ -0,0 +1,95 @@
+package routing
+
+import (
+	"time"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// ActivePayment is a wrapper around a ControlTower that is used to handle its
+// state transitions in the DB.
+type ActivePayment struct {
+	control channeldb.ControlTower
+
+	// inFlight indicates whether this payment is already in-flight. This
+	// is set after the first attempt is registered with the control tower,
+	// or if this is a payment fetcehd from the DB after a restart.
+	inFlight bool
```

Comment:
With the code as it is, `inFlight` is not needed and can be replaced by `Attempt != nil`?

Reply:
n/a anymore